### PR TITLE
Fix typos in compiler/{parser,passes,resolution,util}

### PR DESCRIPTION
**Pull request by cassella**
Including fixing up a few .good files depending on misspelled error messages.

### CODE REVIEW

File: compiler/parser/flex-chapel.cpp [hunk 1]
```diff
@@ -3135,7 +3135,7 @@ static char* eatStringLiteral(yyscan_t scanner, const char* startChar) {
       ParserContext context(scanner);
 
       yyText[0] = '\0';
-      yyerror(yyLloc, &context, "end-of-line in a string literal without a preceeding backslash");
+      yyerror(yyLloc, &context, "end-of-line in a string literal without a preceding backslash");
```
Comment (lydia-duncan) [inline compiler/parser/flex-chapel.cpp]: Note that this is a generated file.  Did you run "make parser" in the parser directory to get these changes?

**Comment by cassella** [comment]
Hmm.  I thought of that while looking the commit over.  I'd only remembered making that change once, so figured it must have "just happened" when I did a full make.  I didn't have bison installed, so that probably wasn't it.  Rebuilding now to test.

**Comment by cassella** [comment]
The versions of flex/bison (2.5.39/3.0.2) available on my machine are much newer than the ones that generated the existing files, and rewrite large portions of the generated files.  Should I submit those (after testing), or would it be better if someone with the old version did it?  (Or revert the pr?)

**Comment by lydia-duncan** [comment]
The version you're using to generate it doesn't matter much, if at all, iirc.  @noakesmichael might recall something I don't, though.  If I'm right, it's fine to just submit the generated files as they are - we try to keep those updates in separate commits so that reviewers don't feel the need to look at them in-depth.  A reviewer for that commit would want to make sure a full std/ run passed for them as well, but other than that wouldn't have to worry much.

**Comment by cassella** [comment]
Sorry about that.


 (And for replying to your line comment with a top-level comment.)

**Comment by lydia-duncan** [comment]
You cad!  You cur!

Just kidding.  It's an easy thing to miss when you're not familiar with the compiler.

**Pull request merged**